GVFS/GVFS/CommandLine/UpgradeVerb.cs

```diff
@@ -94,6 +94,15 @@ private bool TryInitializeUpgrader(out string error)
                 error = null;
                 if (this.upgrader == null)
                 {
+                    string createDirectoryError;
+                    if (!this.fileSystem.TryCreateDirectoryWithAdminAndUserModifyPermissions(ProductUpgraderInfo.GetLogDirectoryPath(), out createDirectoryError))

```

Comment:
@alameenshah this may be related to the issue you wanted to look at.  Should we exit and write error if we fail due to permissions?